tetris.py

```diff
@@ -27,64 +28,162 @@ def run_gui():
     root = tkinter.Tk()
     root.resizable(False, False)
 
-    topbar = tkinter.Label(root)
-    topbar["bg"] = D_GREY
-    topbar.grid(row=0, column=0, sticky="we")
-    game_canvas = tkinter.Canvas(
+    main_canvas = tkinter.Canvas(
         root,
         width=square_size * game_width,
         height=square_size * game_height,
-        highlightthickness=0,
     )
-    game_canvas.grid()
+    main_canvas.grid(row=1, sticky='nswe')
 
-    tetris_gui = TetrisGUI(game_speed, game_canvas)
+    topbar = tkinter.Frame(root, bg=D_GREY, relief = 'ridge')
+    topbar.grid(row=0, columnspan=2, sticky='we')
 
-    root.bind("<Left>", tetris_gui.left_mediator)
-    root.bind("<Right>", tetris_gui.right_mediator)
-    root.bind("<Up>", tetris_gui.rotate_mediator)
+    topbar_time = tkinter.Label(topbar, bg=D_GREY, text='00:00', font = 'digital-7', fg='orange', borderwidth=1)
+    topbar_time.pack(side='left', padx=10)
 
-    tetris_gui.draw_board()
-    tetris_gui.draw_block()
-    tetris_gui.block_mediator()
+    topbar_score = tkinter.Label(topbar, bg=D_GREY, text='foo', font = 'digital-7', fg='orange', borderwidth=1)
+    topbar_score.pack(side='left', fill='x', expand=True)
+
+    topbar_canvas = tkinter.Canvas(topbar, bg=D_GREY, width = square_size*4, height = square_size*2, highlightthickness=0)
+    topbar_canvas.pack(side='right', expand=True)
+
+    sidebar = tkinter.Frame(root, bg=D_GREY)
+    sidebar.grid(row=1, column=1, sticky='nsw')
+
+    new_game_button = tkinter.Button(sidebar, text = 'start')
+    new_game_button.grid(sticky='n')
+
+    new_game_button2 = tkinter.Button(sidebar, text = 'start')
+    new_game_button2.grid(sticky='n')
+
+    new_game_button3 = tkinter.Button(sidebar, text = 'start')
+    new_game_button3.grid(sticky='n')
+
+    tetris_gui = TetrisGUI(game_speed, main_canvas)
+
+    main_board = Board(main_canvas, GREY, game_width, game_height, random.choice(shape_names), (int(game_width / 2), -2), 0)
+    topbar_board = Board(topbar_canvas, D_GREY, square_size*4, square_size*2, random.choice(shape_names), (2, 1), 0)
```

Comment:
Now `tetris_gui` needs to take `main_board` instead of `main_canvas`, so create it after the boards and then pass in the board:

```suggestion
    main_board = Board(main_canvas, GREY, game_width, game_height, random.choice(shape_names), (int(game_width / 2), -2), 0)
    topbar_board = Board(topbar_canvas, D_GREY, square_size*4, square_size*2, random.choice(shape_names), (2, 1), 0)
    tetris_gui = TetrisGUI(game_speed, main_board)
```